### ISSUE TITLE
`kubectl get --watch` exits after ~90 seconds when getting CRDs

### ISSUE THREAD

**Issue by mkuratczyk** [kubernetes/kubectl#668]
I'm not sure whether the problem is on the `kubectl` side or the API server side but reporting here since you can hit this problem when trying to use `kubectl`.

**PROBLEM**
If you run `kubectl get --watch someCRD`, kubectl will exit after some time (between 1 and 2 minutes in my testing). For example:
```
$ time ./_output/bin/kubectl get prometheuses.monitoring.coreos.com --watch

real	1m5.917s
user	0m0.138s
sys	0m0.169s

$ echo $?
0
```

**EXPECTED BEHAVIOUR**
`kubectl` should wait (watch) indefinitely as it does for built-in resources types. On the same machine, I can run `kubectl get --watch pods` or `kubectl get --watch services` and it doesn't exit until I hit ctrl-c.

**VERSIONS**
I've reproduced this problem on a few machines using different kubectl and server versions as well as using different CRDs. This is one of the combinations:
```
$ ./_output/bin/kubectl version
Client Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.0-alpha.0.922+56b40066d55d08", GitCommit:"56b40066d55d08308384b20dfbc292520cb1618a", GitTreeState:"clean", BuildDate:"2019-06-12T10:54:48Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-17T00:58:35Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by seans3** [comment]
/kind bug
/sig cli
/area kubectl

Giving an initial priority of P2. Please update the priority if necessary.

/priority P2

**Comment by seans3** [comment]
Please note the client/server versions are out of the support window. Kubernetes only supports +/- 1 version for client/server. So the 1.14 API server is only guaranteed to work with 1.13, 1.14, and 1.15 versions of kubectl. Please update the issue with versions that are in the support window. Otherwise, the issue will eventually be closed.

**Comment by mkuratczyk** [comment]
I've reproduced it using kubectl 1.14.2 on a 1.14.2 cluster (Kind) as well as 1.13.5 (GKE).

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-16T18:55:03Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-17T00:58:35Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"linux/amd64"}
```
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.2", GitCommit:"66049e3b21efe110454d67df4fa62b08ea79a19b", GitTreeState:"clean", BuildDate:"2019-05-16T18:55:03Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:19:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Comment by seans3** [comment]
Thanks Michal for the effort. Please let me know if you're comfortable with the priority P2.

**Comment by mkuratczyk** [comment]
Yes, P2 is fine. I'm also happy to do some further debugging but with a quick look at kubectl code I could not find any reason why it would treat CRDs differently than other resources. If you have any guesses where the problem can be / pointers at how to debug this kind of issue, I'm happy to spend some more time on it.

Thank you,

**Comment by pswica** [comment]
I found I got the same issue so I tried to address in the PR above.

**Comment by pswica** [comment]
A better fix than mine was created here:
https://github.com/kubernetes/kubernetes/pull/78962
It should get resolved in v1.16
